udata/auth/views.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+
+from flask import Blueprint, current_app, request, redirect
+from udata.auth import current_user
+
+auth = Blueprint('auth', __name__)
+
+@auth.before_app_request
+def ensure_https_authenticated_users():
+    # Force authenticated users to use https
+    if (not current_app.config.get('TESTING', False)
+            and current_app.config.get('USE_SSL', False)
+            and current_user.is_authenticated()
+            and not request.is_secure):
+        return redirect(request.url.replace('http://', 'https://'))
```

Comment:
Yes, and because it's very tricky to do that from the reverse proxy and very easy from the backend which know when the user is authenticated.
The other alternative is to always redirect to https (which can be done from the reverse proxy)